apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -406,12 +409,17 @@ def _cs_reader(
         ):
             major_coords = indices[self.major_axis]
             minor_coords = indices[self.minor_axis]
-            cls = sparse.csr_matrix if self.major_axis == 0 else sparse.csc_matrix
-            sp = cls(
-                sparse.coo_matrix(
-                    (d, (i, j)), shape=self._mk_shape(major_coords, minor_coords)
-                )
-            )
+            shape = self._mk_shape(major_coords, minor_coords)
+            assert self.context is not None
+            sp = CompressedMatrix.from_ijd(
+                i,
+                j,
+                d,
+                shape,
+                "csr" if self.major_axis == 0 else "csc",
+                True,
+                self.context,
+            ).to_scipy()
```

Comment:
To make it easier to read

```suggestion
            sp = CompressedMatrix.from_ijd(
                i,
                j,
                d,
                shape=shape,
                format="csr" if self.major_axis == 0 else "csc",
                make_sorted=True,
                context=self.context,
            ).to_scipy()
```